### PR TITLE
Make terminal width facility compatible with windows, linux, macOS

### PR DESCRIPTION
**Pull request by joshiayush**
* add `struct window_size` to hold the row and column number of the current terminal
* add function `get_terminal_dimensions()` to fill the `window_size` object with terminal
	dimensions
* make function `get_terminal_dimesions()` compatible with `windows`, `macOS`, `linux` and `BSD`
	systems
* include the above facility inside `util` namespace

@jbms Please review my changes.

### CODE REVIEW

**Comment by joshiayush** [comment]
> Also, does the `ioctl` not work on cygwin? I would suppose that on cygwin you might be using something other than the
>  normal command prompt and then would `GetConsoleScreenBufferInfo` still work?

@jbms  I've answered your question [here](https://github.com/jbms/argparse/pull/3#discussion_r714500177)

File: src/jbms/argparse.cpp [hunk 1]
```diff
@@ -5,11 +5,13 @@
 #include <cctype>
 #include <iostream>
 
-// Needed to determine terminal width
-// TODO: add support for Windows
-#include <sys/ioctl.h>
-#include <stdio.h>
-#include <unistd.h>
+#if defined(_WIN32) || (defined(__CYGWIN__) \
+    && !defined(_WIN32))
```
Comment (jbms) [inline src/jbms/argparse.cpp]: What purpose is this `!defined(_WIN32)` condition serving?

**Comment by jbms** [comment]
Thanks for your updates.  Please just update this pull request rather than creating a new pull request.

File: src/jbms/argparse.cpp (hunk 1 above)
Comment (joshiayush) [reply src/jbms/argparse.cpp]: that's a mistake, let me update the PR.

Comment (joshiayush) [reply src/jbms/argparse.cpp]: @jbms although that's a mistake but it's just a pre-proccessor directive we are not going to get any kind of performance hit. Do you want me to update the PR or just merge it and we will remove it later?

Comment (joshiayush) [reply src/jbms/argparse.cpp]: I'm waiting....

**Pull request closed without merging**